js/style/style.js

```diff
@@ -427,6 +427,37 @@ Style.prototype = util.inherit(Evented, {
         }.bind(this));
     },
 
+    featuresIn: function(bbox, params, callback) {
+        var features = [];
+        var error = null;
+
+        if (params.layer) {
+            params.layer = { id: params.layer };
+        }
+
+        util.asyncEach(Object.keys(this.sources), function(id, callback) {
+            var source = this.sources[id];
+            source.featuresIn(bbox, params, function(err, result) {
+                if (result) features = features.concat(result);
+                if (err) error = err;
+                callback();
+            });
+        }.bind(this), function() {
+            if (error) return callback(error);
+
+            // Question: why not do this beforehand, and avoid querying layers
+            // we're just gonna filter out later?
```

Comment:
The intent of this is not to avoid querying particular layers, it's is mitigation for [a style mutation race condition](https://github.com/mapbox/mapbox-gl-js/issues/1096), where a layer is removed from the style in between when featuresAt is initiated and when the callback is executed. As a preflight check it wouldn't address the race condition.